src/target/renesas.c

```diff
@@ -0,0 +1,938 @@
+ *					  Erase:  1 block
+ */
+#define MF3_CF_BLOCK_SIZE       (0x800U)
+#define MF3_RA2A1_CF_BLOCK_SIZE (0x400U)
+#define MF3_DF_BLOCK_SIZE       (0x400U)
+#define MF3_CF_WRITE_SIZE       (0x40U)
+#define MF3_DF_WRITE_SIZE       (0x1U)
+
+/* RV40 Flash */
+/*
+ * RV40F Flash Memory Specifications
+ * Block Size: Code area: 8 KB/32KB  Data area: 64 Bytes
+ * Program/Erase unit Program: Code area: 128 Bytes, Data area: 4/8/16 Bytes
+ *					  Erase: 1 block
+ */
+#define RV40_CF_REGION0_SIZE       (0x10000UL)
+#define RV40_CF_REGION0_BLOCK_SIZE (0x2000U)
+#define RV40_CF_REGION1_BLOCK_SIZE (0x8000U)
+#define RV40_DF_BLOCK_SIZE         (0x40U)
+#define RV40_CF_WRITE_SIZE         (0x80U)
+#define RV40_DF_WRITE_SIZE         (0x4U)
+
+/* RV40 Flash Commands */
+#define RV40_CMD               0x407E0000UL
+#define RV40_CMD_PROGRAM       (0xE8U)
+#define RV40_CMD_PROGRAM_CF    (0x80U)
+#define RV40_CMD_PROGRAM_DF    (0x02U)
+#define RV40_CMD_BLOCK_ERASE   (0x20U)
+#define RV40_CMD_PE_SUSPEND    (0xB0U)
+#define RV40_CMD_PE_RESUME     (0xD0U)
+#define RV40_CMD_STATUS_CLEAR  (0x50U)
+#define RV40_CMD_FORCED_STOP   (0xB3U)
+#define RV40_CMD_BLANK_CHECK   (0x71U)
+#define RV40_CMD_CONFIG_SET_1  (0x40U)
+#define RV40_CMD_CONFIG_SET_2  (0x08U)
+#define RV40_CMD_LOCK_BIT_PGM  (0x77U)
+#define RV40_CMD_LOCK_BIT_READ (0x71U)
+#define RV40_CMD_FINAL         (0xD0U)
+
+#define RV40_BASE 0x407FE000UL
+
+#define RV40_FASTAT       (RV40_BASE + 0x10U) /* Flash Access Status */
+#define RV40_FASTAT_CMDLK (1U << 4U)          /* Command Lock */
+
+#define RV40_FSTATR (RV40_BASE + 0x80U) /* Flash Status */
+
+#define RV40_FSTATR_DBFULL (1U << 10U) /* Data Buffer Full */
+#define RV40_FSTATR_RDY    (1U << 15U) /* Flash Ready */
+
+#define RV40_FSTATR_PRGERR    (1U << 12U) /* Programming Error */
+#define RV40_FSTATR_ERSERR    (1U << 13U) /* Erasure Error */
+#define RV40_FSTATR_ILGLERR   (1U << 14U) /* Illegal Command Error */
+#define RV40_FSTATR_OTERR     (1U << 20U) /* Other Error */
+#define RV40_FSTATR_SECERR    (1U << 21U) /* Security Error */
+#define RV40_FSTATR_FESETERR  (1U << 22U) /* FENTRY Setting Error */
+#define RV40_FSTATR_ILGCOMERR (1U << 23U) /* Illegal Command Error */
+
+#define RV40_FSADDR (RV40_BASE + 0x30U)
+
+#define RV40_FMEPROT        (RV40_BASE + 0x44U)
+#define RV40_FMEPROT_LOCK   (0xD901U)
+#define RV40_FMEPROT_UNLOCK (0xD900U)
+
+#define RV40_FENTRYR            (RV40_BASE + 0x84U)
+#define RV40_FENTRYR_KEY_OFFSET 8U
+#define RV40_FENTRYR_KEY        (0xAAU << RV40_FENTRYR_KEY_OFFSET)
+#define RV40_FENTRYR_PE_CF      (1U)
+#define RV40_FENTRYR_PE_DF      (1U << 7U)
+
+#define RV40_FCPSR         (RV40_BASE + 0xE0U)
+#define RV40_FCPSR_ESUSPMD 1U
+
+#define RENESAS_CF_END (0x00200000UL) /* End of Flash (maximum possible accross families) */
+
+static bool renesas_uid(target *t, int argc, const char **argv);
+
+const struct command_s renesas_cmd_list[] = {
+	{"uid", (cmd_handler)renesas_uid, "Prints unique number"},
```

Comment:
This cast should be removed - the function signature should match the cmd_handler type